### PR TITLE
Add tribuo sample

### PR DESCRIPTION
**Pull request by johanvos**
Add a basic sample that integrates Tribuo in a mobile app.

### CODE REVIEW

File: hello-tribuo/pom.xml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+    <groupId>com.gluonhq.samples</groupId>
+    <artifactId>hellotribuo</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+
+    <name>Hello Tribuo</name>
+
+    <properties>
```
Comment (jperedadnr) [inline hello-tribuo/pom.xml:21]: These properties (except `main.class`) are already defined by the parent [pom](https://github.com/gluonhq/gluon-samples/blob/master/pom.xml#L12), so they can be removed from here (rename `javafx.plugin.version` to `javafx.maven.plugin.version`)

File: hello-tribuo/pom.xml [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.tribuo</groupId>
+            <artifactId>tribuo-all</artifactId>
+            <version>4.0.1</version>
+            <type>pom</type>
+        </dependency>
+        <dependency>
```
Comment (jperedadnr) [inline hello-tribuo/pom.xml:38]: I know these are generated from Gluon Start, but `javafx-base` and `javafx-graphics` are not needed, they are transitive dependencies from `javafx-controls`

File: hello-tribuo/pom.xml [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                <configuration>
+                    <mainClass>${main.class}</mainClass>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>com.gluonhq</groupId>
+                <artifactId>client-maven-plugin</artifactId>
+                <version>${client.maven.plugin.version}</version>
+                <configuration>
+<verbose>true</verbose>
```
Comment (jperedadnr) [inline hello-tribuo/pom.xml:105]: Needs indentation

File: hello-tribuo/pom.xml [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+
+    <groupId>com.gluonhq.samples</groupId>
+    <artifactId>hellotribuo</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+
+    <name>Hello Tribuo</name>
```
Comment (jperedadnr) [inline hello-tribuo/pom.xml:19]: Probably better to rename to `HelloTribue`? Better to avoid white spaces

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+    Series<String, Number> tpSeries = new Series<>();
+    Series<String, Number> fpSeries = new Series<>();
+
+    @Override
+    public void init() {
+        tpSeries.setName("TP");
+        fpSeries.setName("FP");
+        addViewFactory(HOME_VIEW, () -> {
+            FloatingActionButton fab = new FloatingActionButton(MaterialDesignIcon.SEARCH.text,
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:52]: If it is not used, better remove it. But since it is here, why don't we use it to trigger the `train` call? (It will help with nag screen, and chart animation)

File: hello-tribuo/src/main/resources/com/gluonhq/samples/sample/styles.css [hunk 6]
```diff
@@ -0,0 +1,4 @@
+Label {
+    -fx-font-size: 2em;
```
Comment (jperedadnr) [inline hello-tribuo/src/main/resources/com/gluonhq/samples/sample/styles.css:2]: This is too big, 1.3em should be enough

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+                protected void updateAppBar(AppBar appBar) {
+                    appBar.setTitleText("Gluon Mobile");
+                }
+            };
+
+            fab.showOn(view);
+
+            return view;
+        });
+        train();
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:73]: As mentioned before, better move `train()` to an action, or at least call it when the view is shown, not before.

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+                        double f1 = evaluation.f1(label);
+                        double fn = evaluation.fn(label);
+                        double fp = evaluation.fp(label);
+                        double tn = evaluation.tn(label);
+                        double tp = evaluation.tp(label);
+                        javafx.application.Platform.runLater(() -> {
+                            tpSeries.getData().add(new Data(label.getLabel(), tp));
+                            fpSeries.getData().add(new Data(label.getLabel(), fp));
+                        });
+                        System.err.println("Label " + label + " f1 = " + f1 + ", fn = " + fn + ", fp = " + fp);
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:118]: There are `Logger` imports, but there are not used?

File: hello-tribuo/src/main/resources/bezdekIris.data [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+6.9,3.1,5.1,2.3,Iris-virginica
+5.8,2.7,5.1,1.9,Iris-virginica
+6.8,3.2,5.9,2.3,Iris-virginica
+6.7,3.3,5.7,2.5,Iris-virginica
+6.7,3.0,5.2,2.3,Iris-virginica
+6.3,2.5,5.0,1.9,Iris-virginica
+6.5,3.0,5.2,2.0,Iris-virginica
+6.2,3.4,5.4,2.3,Iris-virginica
+5.9,3.0,5.1,1.8,Iris-virginica
+
```
Comment (jperedadnr) [inline hello-tribuo/src/main/resources/bezdekIris.data:151]: There is an extra empty line that causes a warning.

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+                }
+            }
+        } ;
+        thread.start();
+    }
+
+    private Chart createChart() {
+        CategoryAxis xAxis = new CategoryAxis();
+        NumberAxis yAxis = new NumberAxis();
+        StackedBarChart<String, Number> chart = new StackedBarChart(xAxis, yAxis);
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:129]: `new StackedBarChart<>(xAxis, yAxis);`

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+import com.gluonhq.attach.display.DisplayService;
+import com.gluonhq.attach.util.Platform;
+import com.gluonhq.charm.glisten.application.MobileApplication;
+import com.gluonhq.charm.glisten.control.AppBar;
+import com.gluonhq.charm.glisten.control.FloatingActionButton;
+import com.gluonhq.charm.glisten.mvc.View;
+import com.gluonhq.charm.glisten.visual.MaterialDesignIcon;
+import com.gluonhq.charm.glisten.visual.Swatch;
+import java.net.URL;
+import java.util.Set;
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:12]: There are some unused imports

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+        fpSeries.setName("FP");
+        addViewFactory(HOME_VIEW, () -> {
+            FloatingActionButton fab = new FloatingActionButton(MaterialDesignIcon.SEARCH.text,
+                    e -> train());
+
+            Chart chart = createChart();
+
+            Label label = new Label("Hello, Tribuo with Gluon Mobile!");
+
+            VBox root = new VBox(20, chart, label);
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:59]: place `label` on top, so it doesn't get overlapped by the floating button

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+import org.tribuo.classification.evaluation.LabelEvaluator;
+import org.tribuo.classification.sgd.linear.LogisticRegressionTrainer;
+import org.tribuo.common.tree.TreeModel;
+import org.tribuo.data.csv.CSVLoader;
+import org.tribuo.datasource.ListDataSource;
+import org.tribuo.evaluation.TrainTestSplitter;
+
+public class Main extends MobileApplication {
+
+    Series<String, Number> tpSeries = new Series<>();
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:44]: Add a comment with a small description about what tp and fp stands for?

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+
+    private Chart createChart() {
+        CategoryAxis xAxis = new CategoryAxis();
+        NumberAxis yAxis = new NumberAxis();
+        StackedBarChart<String, Number> chart = new StackedBarChart(xAxis, yAxis);
+        ObservableList<XYChart.Series<String, Number>> chartData = FXCollections.observableArrayList();
+        chartData.add(tpSeries);
+        chartData.add(fpSeries);
+        chart.setData(chartData);
+        return chart;
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:134]: Adding `chat.setAnimated(false)` removes the glitches with the axes layout.

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+            Dimension2D dimension2D = DisplayService.create()
+                    .map(DisplayService::getDefaultDimensions)
+                    .orElse(new Dimension2D(640, 480));
+            scene.getWindow().setWidth(dimension2D.getWidth());
+            scene.getWindow().setHeight(dimension2D.getHeight());
+        }
+    }
+
+    private void train() {
+        Thread thread = new Thread(){
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:90]: Before starting the thread, add:
```
tpSeries.getData().clear();
fpSeries.getData().clear();
```        
to clear the series every time `train()` is called?

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+                    var irisHeaders = new String[]{"sepalLength", "sepalWidth", "petalLength", "petalWidth", "species"};
+                    ListDataSource<org.tribuo.classification.Label> irisData
+                            = new CSVLoader<>(new LabelFactory()).loadDataSource(dataUrl, irisHeaders[4], irisHeaders);
+                    TrainTestSplitter<org.tribuo.classification.Label> irisSplitter = new TrainTestSplitter<>(irisData, 0.7, 1L);
+                    MutableDataset<org.tribuo.classification.Label> trainData = new MutableDataset<>(irisSplitter.getTrain());
+                    MutableDataset<org.tribuo.classification.Label> testData = new MutableDataset<>(irisSplitter.getTest());
+                    var cartTrainer = new CARTClassificationTrainer();
+                    TreeModel<org.tribuo.classification.Label> tree = cartTrainer.train(trainData);
+
+                    var linearTrainer = new LogisticRegressionTrainer();
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:103]: Unused

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+
+                    for (org.tribuo.classification.Label label : trainData.getOutputs()) {
+                        double f1 = evaluation.f1(label);
+                        double fn = evaluation.fn(label);
+                        double fp = evaluation.fp(label);
+                        double tn = evaluation.tn(label);
+                        double tp = evaluation.tp(label);
+                        javafx.application.Platform.runLater(() -> {
+                            tpSeries.getData().add(new Data(label.getLabel(), tp));
+                            fpSeries.getData().add(new Data(label.getLabel(), fp));
```
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:115]: `new Data<>(...);`

File: hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java (hunk 14 above)
Comment (jperedadnr) [inline hello-tribuo/src/main/java/com/gluonhq/samples/sample/Main.java:134]: You can add a title?

File: hello-tribuo/src/main/resources/bezdekIris.data (hunk 9 above)
Comment (johanvos) [reply hello-tribuo/src/main/resources/bezdekIris.data:151]: It's the official data set, so I'd like to keep that.

**Review by jperedadnr — APPROVED** [review]
